lua/possession/query.lua

```diff
@@ -30,6 +30,19 @@ function M.filter_by(sessions, opts)
     end, sessions)
 end
 
+--- Produces a list of distinct values
+---@param input table[]
+function M.distinct_list(input)
```

Comment:
If you think this is a premature abstraction, let me know. Happy to undo it since it's only used in that one place currently.

Reply:
I think it would be better to put this in `utils.lua` as this is a generic function not related to sessions.

Also, this implementation has O(n^2) complexity (O(n) for `pairs` * O(n) for linear scan in `vim.tbl_contains`). Besides this, you should generally use `ipairs` for list-like tables. 

A simple O(n) solution is `vim.tbl_keys(utils.list_to_set(list))` (O(n) for `list_to_set` **+** O(n) for `vim.tbl_keys`), but it doesn't preserve order. An improved and optimized version (avoids 2 O(n) loops) which requires more code would be:
```lua
function M.distinct_list(input)
    local unique = {}
    local output = {}
    for _, v in ipairs(input) do
        if not unique[v] then
            unique[v] = true
            table.insert(output, v)
        end
    end
    return output
end
```

As a side note, `unique_list` sounds more familiar to me than `distinct_list`, but it may be my engilsh. 

Sorry if I'm nitpicking too much :P

Reply:
Nitpicking is perfectly fine, especially since you are the maintainer. I've used your improved implementation, moved it to `utils` with a better name and hopefully correct type specs.

I need to leave it as `pair` since the data is in the form `percent_encoded_name => decoded_name`, it's not a list. I think my original name was responsible for part of the confusion since it was called `distinct_list`. It created a list, it didn't take one. It was a poor name.

Reply:
As we have changed the behaviour of `PossessionLoadCwd` this function is no longer required, and has been removed.